tools/build_tar/buildtar.go

```diff
@@ -276,6 +276,13 @@ func (f *tarFile) addTar(toAdd string) error {
 	root := ""
 	if f.directory != "/" {
 		root = f.directory
+		header := tar.Header{
```

Comment:
I think you may need to also specify other parameters like uid, gid, uname, gname, and mode, like here:
https://github.com/kubernetes/repo-infra/blob/0819ae34d1f0332d069a0a022a7504614e7b8422/tools/build_tar/buildtar.go#L217-L225

Otherwise these directories end up being unusable:
```
$ tar tvf bazel-bin/build/debs/kubernetes-cni-data.tar
tar: Removing leading `/' from member names
d--------- 0/0               0 1969-12-31 16:00 /opt/
d--------- 0/0               0 1969-12-31 16:00 /opt/cni/
drwxr-xr-x root/root         0 2017-08-17 10:03 /opt/cni/bin/
-rwxr-xr-x root/root   2814104 2017-08-17 10:02 /opt/cni/bin/flannel
-rwxr-xr-x root/root   3877986 2017-08-17 10:03 /opt/cni/bin/ptp
-rwxr-xr-x root/root   2991965 2017-08-17 10:03 /opt/cni/bin/host-local
-rwxr-xr-x root/root   3470464 2017-08-17 10:02 /opt/cni/bin/portmap
-rwxr-xr-x root/root   2808402 2017-08-17 10:02 /opt/cni/bin/tuning
-rwxr-xr-x root/root   3475750 2017-08-17 10:03 /opt/cni/bin/vlan
-rwxr-xr-x root/root   2605279 2017-08-17 10:03 /opt/cni/bin/sample
-rwxr-xr-x root/root   9921982 2017-08-17 10:03 /opt/cni/bin/dhcp
-rwxr-xr-x root/root   3475802 2017-08-17 10:02 /opt/cni/bin/ipvlan
-rwxr-xr-x root/root   3520724 2017-08-17 10:02 /opt/cni/bin/macvlan
-rwxr-xr-x root/root   3026388 2017-08-17 10:02 /opt/cni/bin/loopback
-rwxr-xr-x root/root   3890407 2017-08-17 10:02 /opt/cni/bin/bridge
```